src/ui/components/instantiate/Step1.tsx

```diff
@@ -35,11 +35,14 @@ export function Step1() {
     ...metadataValidation
   } = useMetadataField();
 
-  useEffect((): void => {
-    if (metadataValidation.name) {
-      setName(metadataValidation.name);
-    }
-  }, [metadataValidation.name, setName]);
+  useEffect(
+    function updateNameFromMetadata(): void {
+      if (metadataValidation.name && nameValidation.isEmpty) {
```

Comment:
instead of creating a new validation field you can just try this:
```
  useEffect(
    function updateNameFromMetadata(): void {
      if (metadataValidation.name && !name && !nameValidation.isTouched) {
        setName(metadataValidation.name);
      }
    },
    [metadataValidation.name, name, nameValidation.isTouched, setName]
  );
```

Reply:
Was my first implementation as well but it's get ugly quick as `!" "` would also be `true`. So you actually need to check the `name.trim().length === 0"` at best. But yes I'm fine either way. 

Let me know if you want me to change it to your check.

Reply:
Refactored the code to this simpler approach.

![demo](https://user-images.githubusercontent.com/839848/225617274-69e833a7-064f-4faa-a7bd-442ad56924f9.gif)